packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx

```diff
@@ -42,57 +51,116 @@ export default function ValidationErrorConsole({ errors, font }: { errors?: IErr
+
+			if (dragging && e.clientY < maxHeight && e.clientY > minHeight) {
+				const deltaY = initialY - e.clientY;
+				const newHeight = initialHeight + deltaY;
+				setHeight(newHeight);
+			}
+		};
+
+		window.addEventListener("mouseup", onMouseUp);
+		window.addEventListener("mousemove", onMouseMove);
+
+		return () => {
+			window.removeEventListener("mouseup", onMouseUp);
+			window.removeEventListener("mousemove", onMouseMove);
+		};
+	}, [state]);
+
 	return (
 		<div
-			className={`absolute bottom-0 left-0 z-10 ${
-				isOpenErrorConsole ? "h-[120px]" : "h-[37px]"
-			} w-full border-t-1 border-subtle bg-default pb-1 pt-1 transition-all`}
+			ref={validationConsoleDiv}
+			style={{
+				height: isOpenErrorConsole ? `${height}px` : `37px`,
+				cursor: isOpenErrorConsole ? `row-resize` : `auto`,
+				paddingTop: isOpenErrorConsole ? `8px` : `0px`,
+				transition: state.current.dragging ? "none" : "height 0.2s ease-out",
+			}}
+			onMouseDown={(e) => {
+				if (e.button === 0 && isOpenErrorConsole) {
+					state.current.initialY = e.clientY;
+					state.current.initialHeight = height;
+					state.current.dragging = true;
+				}
+			}}
+			className="absolute bottom-0 left-0 bg-transparent w-full select-none overflow-hidden"
 		>
-			<div className="flex flex-col h-full">
-				<div className="flex items-center">
-					<ErrorAndWarningCounter
-						errorsCount={errorCount}
-						warningsCount={0}
-						isOpen={isOpenErrorConsole}
-						setOpen={setIsOpenErrorConsole}
-					/>
-					<ErrorAndWarningCounter
-						isWarning
-						errorsCount={0}
-						warningsCount={warningsCount}
-						isOpen={isOpenErrorConsole}
-						setOpen={setIsOpenErrorConsole}
-					/>
-				</div>
-				{isOpenErrorConsole && (
-					<div className="mt-2 flex h-[calc(100%-45px)] flex-col gap-y-1 overflow-auto px-[25px]">
-						{errors?.customWarnings &&
-							errors.customWarnings?.length > 0 &&
-							errors.customWarnings.map((warning: string, index: number) => {
-								return <ErrorMessage key={index} customWarnings={warning} font={font} />;
-							})}
-						{serverSideValidationResult.result?.error && (
-							<ErrorMessage
-								serverSideError={
-									serverSideValidationResult.result?.message + " - " + serverSideValidationResult.result?.error
-								}
-								font={font}
-							/>
-						)}
-						{errors?.ajvErrors &&
-							errors.ajvErrors?.length > 0 &&
-							errors.ajvErrors.map((error: IAjvError, index: number) => {
-								return <ErrorMessage key={index} ajvError={error} font={font} />;
-							})}
-						{errors?.customErrors &&
-							errors.customErrors?.length > 0 &&
-							errors.customErrors.map((error: string, index: number) => {
-								return <ErrorMessage key={index} customErrors={error} font={font} />;
-							})}
-						{errors?.jsYamlError?.mark?.line && <ErrorMessage jsYamlError={errors?.jsYamlError} font={font} />}
+			<AutoSizer>
+				{({ height }) => (
```

Comment:
**Question:** Why do we need `AutoSizer` here?

Reply:
@bripkens at initial implementation I used it to fix transition and lag problem. Now removed it and pushed. thanks for notice. @bripkens